File: output/llama_4_maverick/avatar/algo_based_translation/python/java/codeforces_637_A.java

import java.util.*;

public class codeforces_637_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();
        String[] s = sc.nextLine().trim().split(" ");
        Map<String, Integer> w = new HashMap<>();
        for (String str : s) {
            w.put(str, w.getOrDefault(str, 0) + 1);
        }
        List<String> list = Arrays.asList(s);
        Collections.reverse(list);
        int c = -1;
        String a = "";
        int maxFreq = Collections.max(w.values());
        for (String i : w.keySet()) {
            if (w.get(i) == maxFreq) {
                if (list.indexOf(i) > c) {
                    a = i;
                    c = list.indexOf(i);
                }
            }
        }
        System.out.println(a);
    }
}

File: output/llama_4_maverick/avatar/algo_based_translation/java/python/successful_translation/codeforces_637_A.py

def main():
    n = int(input())
    ara = list(map(int, input().split()))
    freq_map = {}
    for v in ara:
        if v not in freq_map:
            freq_map[v] = 1
        else:
            freq_map[v] += 1
    max_freq = max(freq_map.values())
    mm = {}
    for v in ara:
        if v not in mm:
            mm[v] = 1
        else:
            mm[v] += 1
        if mm[v] == max_freq:
            print(v)
            break

if __name__ == "__main__":
    main()

